packages/@react-spectrum/s2/src/Card.tsx

```diff
@@ -382,6 +382,7 @@ export const Card = forwardRef(function Card(props: CardProps, ref: DOMRef<HTMLD
   [props] = useSpectrumContextProps(props, ref, CardContext);
   let domRef = useDOMRef(ref);
   let {density = 'regular', size = 'M', variant = 'primary', UNSAFE_className = '', UNSAFE_style, styles, id, ...otherProps} = props;
+  id = useId(props.id ? String(props.id) : undefined);
```

Comment:
Do we need the `useId`? Could just do `id={props.id != null ? String(props.id) : undefined}` (null check as well in case `id` is zero.

Reply:
makes sense